src/Microsoft.Performance.SDK/Extensibility/ICookedDataRetrievalExtensions.cs

```diff
@@ -0,0 +1,202 @@
+        ///     Unique identifier for the data to retrieve.
+        /// </param>
+        /// <param name="result">
+        ///     The uniquely identified data.
+        /// </param>
+        /// <exception cref="InvalidOperationException">
+        ///     The specified <paramref name="dataOutputPath"/> was not found.
+        /// </exception>
+        /// <exception cref="InvalidCastException">
+        ///     The value at <paramref name="dataOutputPath"/> cannot be cast to <typeparamref name="T"/>.
+        /// </exception>
+        public static void QueryOutput<T>(
+            this ICookedDataRetrieval retrieval,
+            DataOutputPath dataOutputPath,
+            out T result)
+        {
+            result = retrieval.QueryOutput<T>(dataOutputPath);
+        }
+
+        /// <summary>
+        ///     Retrieves data by name.
+        /// </summary>
+        /// <param name="retrieval">
+        ///     Instance on which to act.
+        /// </param>
+        /// <param name="cookerPath">
+        ///     Identifies the data cooker to query.
+        /// </param>
+        /// <param name="outputId">
+        ///     Identifies which data on the data cooker to retrieve.
+        /// </param>
+        /// <returns>
+        ///     The uniquely identified data.
+        /// </returns>
+        /// <exception cref="InvalidOperationException">
+        ///     The specified <paramref name="outputId"/> was not found on the data cooker 
+        ///     <paramref name="cookerPath"/>.
+        /// </exception>
+        public static object QueryOutput(
+            this ICookedDataRetrieval retrieval,
+            DataCookerPath cookerPath,
+            string outputId)
+        {
+            return retrieval.QueryOutput(new DataOutputPath(cookerPath, outputId));
+        }
+
+        /// <summary>
+        ///     Retrieves data by name.
+        /// </summary>
+        /// <param name="retrieval">
+        ///     Instance on which to act.
+        /// </param>
+        /// <param name="cookerPath">
+        ///     Identifies the data cooker to query.
+        /// </param>
+        /// <param name="outputId">
+        ///     Identifies which data on the data cooker to retrieve.
+        /// </param>
+        /// <param name="result">
+        ///     The uniquely identified data.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the value was successfully retrieved; <c>false</c> otherwise.
+        /// </returns>
+        public static bool TryQueryOutput(
+            this ICookedDataRetrieval retrieval,
+            DataCookerPath cookerPath,
+            string outputId,
+            out object result)
+        {
+            return retrieval.TryQueryOutput(new DataOutputPath(cookerPath, outputId), out result);
+        }
+
+        /// <summary>
+        ///     Retrieves data by name, typecast to the given type.
+        ///     The caller is coupled to the data extension and expected to know the
```

Comment:
what exactly does "The caller is coupled to the data extension" mean? Is this just saying to call this generic method the caller needs to know at compile time what T is? Isn't that implied with a typeparam in the signature?

Reply:
It means that it must know the return type - there can be no compile time type check. And if that return type changes, the caller must be updated.